servers/server.js

```diff
@@ -0,0 +1,27 @@
+const http = require('http');
+const url = require('url');
+const airport = require('../src/airportDetailsRetriever');
+
+const host = 'localhost';
+const port = 3000;
+
+const server = http.createServer((req, res) => {
+  const queries = url.parse(req.url, true).query;
+  
+  if ('airport-code' in queries) {
+    try {
+      const airportDetails = airport.getDetails(queries['airport-code']);
+      res.setHeader('Content-Type', 'application/json');
+      res.writeHead(200);
+      res.end(JSON.stringify(airportDetails, null, '\t'));
+    } catch (err) {
+      res.setHeader('Content-Type', 'text/plain');
+      res.writeHead(400);
```

Comment:
Is it always going to be a bad request client error for all error cases? What if the `airports.json` file becomes unusable tomorrow for some reason, will it still be client side error? Throw 400 when it is indeed a bad request from client. Make `airport.getDetails` throw appropriate exceptions for input validation failure and use that to detect bad request.